### PR TITLE
issue-17 task-c: Creation of DB connectors

### PR DESCRIPTION
**Pull request by a-ajith**
In this PR reverting back to v6 of go-redis fro v8 as it requires Go1.14
Also added connector functions for Create, GetAllKeys and Get

### CODE REVIEW

**Review by shivacharanms — APPROVED** [review]

**Review by bharath-b-hpe — APPROVED** [review]

**Review by lakkundipooja — APPROVED** [review]

File: db/operations.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		return nil
+	}
+}
+
+// GetAllKeys will collect all the keys of provided table
+func (c *Client) GetAllKeys(table string) ([]string, error) {
+	var allKeys []string
+	var cursor uint64
+	for {
+		keys, c, err := c.pool.Scan(cursor, generateKey(table, "*"), 100).Result()
```
Comment (jeevan-kamkar) [inline db/operations.go]: please define value 100 in constant

**Review by jeevan-kamkar — CHANGES_REQUESTED** [review]
Address the changes

File: db/operations.go (hunk 1 above)
Comment (a-ajith) [reply db/operations.go]: addressed

**Review by jeevan-kamkar — APPROVED** [review]

File: db/operations.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+		return nil
+	}
+}
+
+// GetAllKeys will collect all the keys of provided table
+func (c *Client) GetAllKeys(table string) ([]string, error) {
+	var allKeys []string
+	var cursor uint64
+	for {
+		keys, c, err := c.pool.Scan(cursor, generateKey(table, "*"), scanPaginationSize).Result()
```
Comment (Bharath-KKB) [inline db/operations.go]: why do we have to retrieve all keys ? ever ?

**Review by Bharath-KKB — CHANGES_REQUESTED** [review]
should not scan all keys by default. We need sufficient justification to do this.

**Review by Bharath-KKB — APPROVED** [review]

**Review by Bharath-KKB — CHANGES_REQUESTED** [review]
Just saw the last commit and approved. However realised there are 2 previous comments which had commments so cancelled the approval.

File: db/operations.go (hunk 2 above)
Comment (a-ajith) [reply db/operations.go]: @Bharath-KKB This function retrieves all the keys associated with a particular table, which can be used while doing the GET 0n the collections. We are saving the information in DB with "tableName:tableKey" as keys. So. here in this function we are retrieving all keys with the table name and the wild character.

Comment (Bharath-KKB) [reply db/operations.go]: I did not ask what this is doing. I said why we are trying to retrieve all keys.

Comment (a-ajith) [reply db/operations.go]: @Bharath-KKB We are not retrieving all the keys, but the keys associated with a particular type, lets say "Fabric", so we are retrieving all fabric keys for fabric collection with Fabric:*. I will change the function name to more meaning full one

Comment (a-ajith) [reply db/operations.go]: Addressed

Comment (Bharath-KKB) [reply db/operations.go]: What are the table names for :
1) Fabric
2) Switches
3) Ports
4) Zones
5) Address Pools
6) Endpoints ?

Comment (a-ajith) [reply db/operations.go]: @Bharath-KKB table names will be added in the next PR. file path to table name constants is https://github.com/a-ajith/PluginCiscoACI/blob/187bd7a7f57f1290b9bdb57d8bb882badfaac9d9/capmodel/common.go#L26

Comment (a-ajith) [reply db/operations.go]: @Bharath-KKB added pattern as a parameter to the GetAllKeysFromTable function along with table name. So now we are able to get all keys associated with table_name:pattern*.

**Pull request merged**